disco/disco.go

```diff
@@ -49,6 +49,17 @@ type Disco struct {
 	Transport http.RoundTripper
 }
 
+// ErrServiceDiscoveryNetworkRequest represents the error that occurs when
+// the service discovery fails for an unknown network problem.
+type ErrServiceDiscoveryNetworkRequest struct {
```

Comment:
Love it. I hope we can all get into the habit of creating our own errors like this for our own repos.